src/System.Net.Sockets/tests/FunctionalTests/Issue14037.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class Issue14037
+    {
+        [OuterLoop]
+        [Fact]
+        public static void AcceptNonBlocking_Success()
+        {
+            Socket listenSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            int port = listenSocket.BindToAnonymousPort(IPAddress.Loopback);
+
+            listenSocket.Blocking = false;
+
+            listenSocket.Listen(5);
+
+            Task t = Task.Run(() => { DoAccept(listenSocket, 5); });
+
+            // Loop, doing connections and pausing between
+            for (int i = 0; i < 5; i++)
+            {
+                Thread.Sleep(50);
+                Socket connectSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+                connectSocket.Connect(listenSocket.LocalEndPoint);
+            }
+
+            t.Wait();
```

Comment:
In debug, it hits an assert in the code.  In release, it incurs an Argument exception (we're confused because we don't expect this state).

I realize there are no XUnit asserts here, but it wasn't clear what to assert.  Basically, the test is that the non-blocking Accepts succeed eventually; if not the test will hang.  I could change the Wait to have a timeout, but that's tricky because if the timeout is too small, it will generate false negatives; if it's too large, it just looks like a hang anyway.  What do we generally do in this kind of situation?